src/test/regress/expected/multi_deparse_function.out

```diff
@@ -0,0 +1,311 @@
+--
+-- Regression tests for deparsing ALTER/DROP FUNCTION Queries
+--
+-- This test implements all the possible queries as of Postgres 11:
+-- 
+-- ALTER FUNCTION name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ]
+--     action [ ... ] [ RESTRICT ]
+-- ALTER FUNCTION name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ]
+--     RENAME TO new_name
+-- ALTER FUNCTION name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ]
+--     OWNER TO { new_owner | CURRENT_USER | SESSION_USER }
+-- ALTER FUNCTION name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ]
+--     SET SCHEMA new_schema
+-- ALTER FUNCTION name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ]
+--     DEPENDS ON EXTENSION extension_name
+-- 
+-- where action is one of:
+-- 
+--     CALLED ON NULL INPUT | RETURNS NULL ON NULL INPUT | STRICT
+--     IMMUTABLE | STABLE | VOLATILE | [ NOT ] LEAKPROOF
+--     [ EXTERNAL ] SECURITY INVOKER | [ EXTERNAL ] SECURITY DEFINER
+--     PARALLEL { UNSAFE | RESTRICTED | SAFE }
+--     COST execution_cost
+--     ROWS result_rows
+--     SET configuration_parameter { TO | = } { value | DEFAULT }
+--     SET configuration_parameter FROM CURRENT
+--     RESET configuration_parameter
+--     RESET ALL
+-- 
+-- DROP FUNCTION [ IF EXISTS ] name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ] [, ...]
+--     [ CASCADE | RESTRICT ]
+SET citus.next_shard_id TO 20020000;
+CREATE SCHEMA function_tests;
+SET search_path TO function_tests;
+SET citus.shard_count TO 4;
+SET client_min_messages TO INFO;
+CREATE FUNCTION deparse_test(text)
+	RETURNS text
+	AS 'citus'
+ 	LANGUAGE C STRICT;
+-- Create a simple function
+CREATE FUNCTION add(integer, integer) RETURNS integer
```

Comment:
Let's use a lot more complicated function in the tests:

a) Use a function name that needs to be quoted
b) Use a schema name that needs to be quoted (see [here](https://github.com/citusdata/citus/blob/master/src/test/regress/sql/multi_schema_support.sql#L773) for examples)
c) Have an argument with DEFAULT value
d) Have OUT argument
e) Have an INOUT argument
f) Have an VARIADIC argument
g) Use a different type than INT
h) Execute some tests with search path `SET search_path TO public` for sanity checks
i) Sanity check with one function using plpgsql instead of sql
j) Sanity check with RETURNS TABLE -- do we support that?

Reply:
maybe one argument is a composite type: `CREATE TYPE intpair AS (x int, y int);
`

Reply:
I completed all of these items except for one.

> h) Execute some tests with search path SET search_path TO public for sanity checks

I do not get what I should be testing here. If I set the search_path TO public before altering a function, I need to specify the original schema name in the query.

Reply:
will punt on `h` for now.